SA/SA.cna

```diff
@@ -175,18 +180,88 @@ alias ldapsearch {
 		return;
 	}
 
-	$attributes = iff(-istrue $3, $3, "");
-	$result_limit = iff(-istrue $4, $4, 0);
-	$scope = iff(-istrue $5, $5, 3);
-	$hostname = iff(-istrue $6, $6, "");
-	$domain = iff(-istrue $7, $7, "");
 
-	$args   = bof_pack($1, "zziizz", $2, $attributes, $result_limit, $scope, $hostname, $domain);
+	$ldapsearch_len = 11;
+	$query_len = strlen($2) + 1;
+	$total_mandatory_len = $ldapsearch_len + $query_len;
+	$base_len = strlen($0);
+
+	if($base_len > $total_mandatory_len){
+		$data = substr($0, $total_mandatory_len);
+		@args = split('--', $data);
+	}
+	else{
+		@args = @();
+	}
+
+	local('$query $attributes $result_limit $scope $hostname $dn $ldaps');
+
+	$query = replaceQuotes($2);
+	$attributes = "*";
+	$result_limit = 0;
+	$scope = 3;
+	$hostname = "";
+	$dn = "";
+	$ldaps = 0;
+
+    for ($i = 1; $i < size(@args); $i++) {
+        $arg = @args[$i];
+		$option = split(' ', $arg)[0];
+		$value = substr($arg, strlen($option));
+		$value = replaceQuotes($value);
+
+        if ($option iswm "attributes") {
+            $attributes = $value;
+        }
+        else if ($option iswm "count") {
+            $count = $value;
```

Comment:
We reference result_limit in what we actually send, meaning count would never be respected